### ISSUE TITLE
RAGEmbedding singleton mode issue

### ISSUE THREAD

**Issue by kangfenmao** [llm-tools/embedJs#188]
### 🐛 Describe the bug

```js
export class RAGEmbedding {
    static singleton;
    static async init(embeddingModel) {
        if (!this.singleton) {
            await embeddingModel.init();
            this.singleton = new RAGEmbedding(embeddingModel);
        }
    }
    static getInstance() {
        return RAGEmbedding.singleton;
    }
    static getEmbedding() {
        return RAGEmbedding.getInstance().embedding;
    }
    static translateChunks(chunks) {
        return chunks.map((chunk) => {
            return {
                pageContent: chunk.pageContent,
                metadata: chunk.metadata,
            };
        });
    }
    embedding;
    constructor(embeddingModel) {
        this.embedding = embeddingModel;
    }
}
//# sourceMappingURL=rag-embedding.js.map
```

The singleton pattern can cause problems when using multiple APIs

API1: openai
API2: ollama

First embedding text with ollama, then, embedding will always use ollama, never use openai

**Comment by adhityan** [comment]
I agree; in hindsight, the singleton abstraction adds no real value. This will be removed in the upcoming release.

**Comment by adhityan** [comment]
Addressed in `0.1.26`
